### PR TITLE
fix: opsi amount handler and counter ocr

### PR DESCRIPTION
**Pull request by SaarChaffee**
- amount handler
- counter ocr should return number
- counter area

### CODE REVIEW

File: module/os_shop/item.py [hunk 1]
```diff
@@ -43,10 +43,10 @@ def ocr(self, image, direct_ocr=False):
         result_list = super().ocr(image, direct_ocr=direct_ocr)
         if isinstance(result_list, list):
             for r in range(len(result_list)):
-                result_list[r] = result_list[r].split('/')
+                result_list[r] = [int(i) for i in result_list[r].split('/')]
             return result_list
```
Comment (hgjazhgj) [inline module/os_shop/item.py:47]: return [[int(j)for j in i.split('/')]for i in result_list]

File: module/os_shop/shop.py [hunk 2]
```diff
@@ -96,29 +97,57 @@ def shop_buy_amount_handler(self, item):
             ScriptError: OCR_SHOP_AMOUNT
         """
         currency = self.get_currency_coins(item)
+        count = min(int(currency // item.price), item.count)
```
Comment (hgjazhgj) [inline module/os_shop/shop.py:100]: currency和item.price不应该是浮点数吧
int间的__truediv__返回的就是int

File: module/os_shop/item.py (hunk 1 above)
Comment (SaarChaffee) [reply module/os_shop/item.py:47]: lme说不缺这点行数

File: module/os_shop/shop.py (hunk 2 above)
Comment (SaarChaffee) [reply module/os_shop/shop.py:100]: 我也不知道为什么要加个 int https://github.com/LmeSzinc/AzurLaneAutoScript/blob/6ac38928bd713fb74abb391b113a0e72a61cdd68/module/shop/clerk.py#L246

File: module/os_shop/item.py (hunk 1 above)
Comment (LmeSzinc) [reply module/os_shop/item.py:47]: 屑，快改，python下
```
for i in range():
    list[i] ...
```
要打屁股的

Comment (SaarChaffee) [reply module/os_shop/item.py:47]: 诶

**Pull request merged**